rust/worker/src/system/receiver.rs

```diff
@@ -64,3 +64,17 @@ impl ChromaError for ChannelError {
         ErrorCodes::Internal
     }
 }
+
+#[derive(Error, Debug)]
+pub enum ChannelRequestError {
```

Comment:
I could be convinced to combine this with the above `ChannelError`, I made this separate because callers of `.send()`will never have to handle the `ResponseError` case

Reply:
I like super granular errors and not having errors that cannot occur from a method!